app/org/sagebionetworks/bridge/dynamodb/DynamoUserConsent3.java

```diff
@@ -0,0 +1,109 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import org.sagebionetworks.bridge.json.BridgeTypeName;
+import org.sagebionetworks.bridge.json.DateTimeToLongSerializer;
+import org.sagebionetworks.bridge.json.DateTimeToPrimitiveLongDeserializer;
+import org.sagebionetworks.bridge.models.accounts.UserConsent;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBRangeKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBVersionAttribute;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+/**
+ * This version of the consent will accept a withdrewOn timestamp, and the range key is set to the signedOn 
+ * value so there can be multiple records, returned in the order they were created by participant.
+ *
+ */
+@DynamoDBTable(tableName = "UserConsent3")
+@BridgeTypeName("UserConsent")
+public class DynamoUserConsent3 implements UserConsent {
+
+    // The only timestamp that can be null is withdrewOn.
+    private String healthCodeStudy;
+    private String healthCode;
+    private String studyIdentifier;
+    private long consentCreatedOn;
+    private long signedOn;
+    private Long withdrewOn;
+    private Long version;
+    
+    @DynamoDBHashKey
+    public String getHealthCodeStudy() {
+        return healthCodeStudy;
+    }
+    public void setHealthCodeStudy(String healthCodeStudy) {
+        this.healthCodeStudy = healthCodeStudy;
+    }
+    public void setHealthCodeStudy(String healthCode, String studyIdentifier) {
+        this.healthCode = healthCode;
+        this.studyIdentifier = studyIdentifier;
+        this.healthCodeStudy = healthCode + ":" + studyIdentifier;
+    }
+    
+    @DynamoDBAttribute
+    public String getHealthCode() {
+        return healthCode;
+    }
+    public void setHealthCode(String healthCode) {
+        setHealthCodeStudy(healthCode, getStudyIdentifier());
+    }
+
+    @Override
+    @DynamoDBAttribute
+    public String getStudyIdentifier() {
+        return studyIdentifier;
+    }
+    public void setStudyIdentifier(String studyIdentifier) {
+        setHealthCodeStudy(getHealthCode(), studyIdentifier);
+    }
```

Comment:
As written, deserializing this from DDB will cause setHealthCodeStudy() to be called twice (once for studyIdentifier, once for healthCode), and this.healthCodeStudy will be called 3 times.

It would probably be cleaner to only have studyIdentifier and healthCode be member variables. getHealthCodeStudy would just always derive from healthCode + ":" + studyIdentifier.

Alternatively, healthCode and studyId don't need to be in DDB. Neither one is useful as an index, and both can be derived from healthCodeStudy. We could just have getHealthCode() and getStudyIdentifier with a DynamoDBIgnore attribute and no setters.